src/main/scala/co/teapot/tempest/server/TempestDBServer.scala

```diff
@@ -14,48 +14,39 @@
 
 package co.teapot.tempest.server
 
-import java.io.File
 import java.{lang, util}
 
 import co.teapot.tempest._
-import co.teapot.tempest.algorithm.MonteCarloPPR
+import co.teapot.tempest.algorithm.MonteCarloPPRTyped
 import co.teapot.tempest.graph._
-import co.teapot.tempest.typedgraph.{SimpleTypedGraph, UnionTypedGraph}
+import co.teapot.tempest.typedgraph.{BipartiteTypedGraph, IntNode, TypedGraphUnion}
 import co.teapot.tempest.util.{CollectionUtil, ConfigLoader, LogUtil}
 import co.teapot.thriftbase.TeapotThriftLauncher
 import org.apache.thrift.TProcessor
 
-import scala.collection.mutable
 import scala.collection.JavaConverters._
+import scala.collection.mutable
 
 /** Given a graph, this thrift server responds to requests about that graph. */
 class TempestDBServer(databaseClient: TempestDatabaseClient, config: TempestDBServerConfig)
-    extends TempestGraphServer(config.graphDirectoryFile) with TempestDBService.Iface {
+    extends TempestGraphServer(databaseClient, config) with TempestDBService.Iface {
 
   type DegreeFilter = collection.Map[DegreeFilterTypes, Int]
-  override def getMultiNodeAttributeAsJSON(graphName: String, nodeIdsJava: util.List[lang.Long], attributeName: String): util.Map[lang.Long, String] = {
-    val nodeIds = CollectionUtil.toScala(nodeIdsJava)
-    val idToJson = databaseClient.getMultiNodeAttributeAsJSON(graphName, nodeIds, attributeName)
-    CollectionUtil.toJavaLongStringMap(idToJson)
+  override def getMultiNodeAttributeAsJSON(nodesJava: util.List[Node], attributeName: String): util.Map[Node, String] = {
+    databaseClient.getMultiNodeAttributeAsJSON(nodesJava.asScala, attributeName).asJava
   }
 
-  def setNodeAttribute(graphName: String, nodeId: Long, attributeName: String, attributeValue: String): Unit =
-    databaseClient.setNodeAttribute(graphName, nodeId, attributeName, attributeValue)
+  def setNodeAttribute(node: Node, attributeName: String, attributeValue: String): Unit =
+    databaseClient.setNodeAttribute(node, attributeName, attributeValue)
 
-  def nodes(graphName: String, sqlClause: String): util.List[lang.Long] =
-    CollectionUtil.toJava(databaseClient.nodeIdsFiltered(graphName, sqlClause))
-
-  def loadEdgeConfig(edgeType: String): EdgeTypeConfig = {
-    val edgeConfigFile = new File(config.graphConfigDirectoryFile, s"$edgeType.yaml")
-    if (! edgeConfigFile.exists()) {
-      throw new UndefinedGraphException(s"Graph config file ${edgeConfigFile.getCanonicalPath} not found")
-    }
-    ConfigLoader.loadConfig[EdgeTypeConfig](edgeConfigFile)
+  def nodes(nodeType: String, sqlClause: String): util.List[Node] = {
+    val nodeIds = databaseClient.nodeIdsMatchingClause(nodeType, sqlClause)
+    (nodeIds map { id => new Node(nodeType, id)}).asJava
   }
 
-  def typedGraph(edgeType: String): SimpleTypedGraph = {
+  def typedGraph(edgeType: String): BipartiteTypedGraph = {
```

Comment:
This is a good question, and the answer is still unclear to me.  `BipartiteTypedGraph` sort-of serves the function your saying; it takes a graph and adds a source and target node type to it.  I could change `DirectedGraph` to require a node type.  But I want someone to be able to analyze a single graph (say Twitter-2010) by just constructing construct a single `DirectedGraph` and not worry about graph types.  

It's possible we should delete DirectedGraphUnion, and move towards only supporting typed graphs for things like union.
We should keep thinking about this going forward, but I think it's reasonable for now to have a separate hierarchy for plain and typed graphs.